internal/controller/rediscluster/rediscluster_controller.go

```diff
@@ -163,9 +164,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			return intctrlutil.RequeueWithError(ctx, err, "")
 		}
 	}
+	// Update the current replica count in the status
+	if err := r.updateCurrentReplicaCount(ctx, instance); err != nil {
+		return intctrlutil.RequeueWithError(ctx, err, "failed to update replica count")
+	}
 
+	// Replace the early return with a status update
 	if !(r.IsStatefulSetReady(ctx, instance.Namespace, instance.Name+"-leader") && r.IsStatefulSetReady(ctx, instance.Namespace, instance.Name+"-follower")) {
-		return intctrlutil.Reconciled()
+		// Update status to show degraded state
+		err = k8sutils.UpdateRedisClusterStatus(ctx, instance,
+			status.RedisClusterDegraded,
+			"StatefulSet not ready",
+			instance.Status.ReadyLeaderReplicas,
+			instance.Status.ReadyFollowerReplicas,
+			r.Dk8sClient)
+		if err != nil {
+			return intctrlutil.RequeueWithError(ctx, err, "")
+		}
+		// Requeue faster when degraded
+		return intctrlutil.RequeueAfter(ctx, time.Second*5, "waiting for statefulsets to be ready")
```

Comment:
We should use return intctrlutil.Reconciled() because we Own(&appsv1.StatefulSet{}), which will trigger a reconciliation as needed when the StatefulSet status changes.